lib/mix/tasks/time.ex

```diff
@@ -13,36 +13,80 @@ defmodule Mix.Tasks.Time do
     Mix.shell().info(time)
   end
 
-  def run([time_zone]) do
+  def run(args) do
     Mix.Task.run "app.start"
 
+    {opts, [time_zone], _} = OptionParser.parse(args, strict: [am_pm: :boolean])
     {:ok, current_date_time_in_time_zone} = DateTime.now(time_zone)
 
-    day_of_year =
-      current_date_time_in_time_zone
-      |> DateTime.to_date()
-      |> Date.day_of_year()
+    output = format_output(current_date_time_in_time_zone, time_zone, opts)
 
-    city =
-      String.split(time_zone, "/")
-      |> Enum.drop(1)
-      |> to_string()
+    Mix.shell().info(output)
+  end
 
-    time =
-      current_date_time_in_time_zone
-      |> DateTime.truncate(:second)
-      |> DateTime.to_time()
-      |> to_string()
+  def format_output(date_time, time_zone, opts \\ []) do
+    time = time(date_time)
+    {time, suffix} = suffix(time, opts)
+    city = city(time_zone)
 
-    output =
-      case day_of_year do
-        1 ->
-          "#{time} Happy New Year #{city}!"
+    case day_of_the_year(date_time) do
+      1 ->
+        time <> suffix <> " Happy New Year #{city}!"
 
-        _ ->
-          "time in #{city}: #{time}"
-      end
+      _ ->
+        "time in #{city}: #{time} " <> suffix
+    end
+  end
 
-    Mix.shell().info(output)
+  defp suffix(time, [am_pm: am_pm]) when am_pm do
+    hour = String.split(time, ":") |> List.first |> String.to_integer
+    {convert_pm(time), suffix(hour)}
+  end
+
+  defp suffix(time, _), do: {time, ""}
+  defp suffix(hour) when hour < 12, do: "am"
+  defp suffix(hour) when hour > 12, do: "pm"
+
+  defp city(time_zone) do
+    String.split(time_zone, "/")
+    |> Enum.drop(1)
+    |> to_string()
+  end
+
+  defp day_of_the_year(date_time) do
+    date_time
+    |> DateTime.to_date()
+    |> Date.day_of_year()
+  end
+
+  defp time(date_time) do
+    date_time
+    |> DateTime.to_naive
+    |> NaiveDateTime.truncate(:second)
+    |> NaiveDateTime.to_time()
+    |> to_string()
```

Comment:
If you don't convert this to string you could use `hour`, `minute` and `seconds` fields on the `Time` struct and not have to split the string etc.